packages/library-base/src/commonMain/kotlin/io/realm/RealmObject.kt

```diff
@@ -74,9 +83,9 @@ internal fun RealmObject.hasSameObjectKey(other: RealmObject?): Boolean {
     }
 
     val thisKey =
-        RealmInterop.realm_object_get_key(this.realmObjectInternal().`$realm$ObjectPointer`!!)
+        RealmInterop.realm_object_get_key(this.asObjectReference()!!.objectPointer)
```

Comment:
Again, if we reworked the above `isManaged` into `asObjectReference()?.<some-method>` then we wouldn't have to handle all the unmanaged state (`!!`, etc.) here.